sdk/js/__tests__/User.spec.js

```diff
@@ -58,29 +58,47 @@ describe('DVCPopulatedUser tests', () => {
         expect(user.isAnonymous).toBe(false)
     })
 
-    it('should NOT throw an error if no user id and no anonymous flag set', () => {
-        const createUser = () => {
-            new DVCPopulatedUser({})
-        }
-        expect(createUser).not.toThrow(expect.any(Error))
+    it('should create an anonymous user if no user id and no anonymous flag set', () => {
+        const user = new DVCPopulatedUser({})
+        expect(uuidValidate(user.user_id)).toBe(true)
+        expect(user.isAnonymous).toBe(true)
     })
 
     it('should create an anonymous user id if isAnonymous is true', () => {
         const newUser = new DVCPopulatedUser({ isAnonymous: true })
         expect(uuidValidate(newUser.user_id)).toBe(true)
+        expect(newUser.isAnonymous).toBe(true)
     })
 
-    it('should throw an error if the user id is an empty string', () => {
+    it('should use provided user_id when isAnonymous is true', () => {
+        const user = new DVCPopulatedUser({ user_id: 'abc123', isAnonymous: true })
+        expect(user.user_id).toBe('abc123')
+        expect(user.isAnonymous).toBe(true)
```

Comment:
how does calling resetuser change this?

Reply:
what do you mean? This is all kinda outside of `resetUser`: 
<img width="800" alt="image" src="https://github.com/user-attachments/assets/fa2364e0-aeac-42e4-8b9a-ded1e5c9e879" />

Reply:
More the concept that if you have an anon user and you're setting a value for the user id - do we retain that or wipe it?

Reply:
more a curiosity thing than anything

Reply:
good point I think that would be an issue right now with how its setup: 
<img width="676" alt="image" src="https://github.com/user-attachments/assets/1eeec3fa-b71e-40d8-8b16-bcb62b996245" />